crypto/threads_none.c

```diff
@@ -133,7 +133,6 @@ void CRYPTO_THREAD_lock_free(CRYPTO_RWLOCK *lock) {
     if (lock == NULL)
         return;
 
-    *(unsigned int *)lock = 0;
```

Comment:
I am confused as to why this causes a memory leak??

Reply:
sorry I was confused. this is just doing kind `memset()`. I've just realized on my way for lunch. closing this PR as invalid.